lightning/src/ln/channelmanager.rs

```diff
@@ -10357,16 +10397,10 @@ where
 			return Err(Bolt12SemanticError::MissingPaths)
 		}
 
-		let node_id = self.get_our_node_id();
-		let expanded_key = &self.inbound_payment_key;
 		let entropy = &*self.entropy_source;
-		let secp_ctx = &self.secp_ctx;
-
 		let nonce = Nonce::from_entropy_source(entropy);
-		let mut builder = OfferBuilder::deriving_signing_pubkey(
-			node_id, expanded_key, nonce, secp_ctx
-		).chain_hash(self.chain_hash);
 
+		let mut builder = self.flow.create_offer_builder(entropy, None, Some(nonce), Vec::new())?;
```

Comment:
Unfortunately this doesn't work as-is because `Flow::create_offer_builder` will set the offer's blinded message paths to contain paths to the recipient node, whereas async receive offers need to *only* contain the parameterized `message_paths_to_always_online_node`. So I think we should probably move this whole method into `flow.rs` to ensure the correct behavior.

Reply:
Makes sense! Moved it to flow in [pr3639.12](https://github.com/shaavan/rust-lightning/commits/pr3639.12)
Thanks for the pointer!